src/core/Akka/Actor/SupervisorStrategy.cs

```diff
@@ -193,10 +190,9 @@ public static SupervisorStrategy DefaultStrategy
     /// </summary>
     public class OneForOneStrategy : SupervisorStrategy
     {
-        private readonly int _maxNumberOfRetries;
-        private readonly int _withinTimeRangeMilliseconds;
-        private readonly Func<Exception, Directive> _decider;
-        private readonly bool _loggingEnabled;
+        public int MaxNumberOfRetries { get; private set; }
+        public int WithinTimeRangeMilliseconds { get; private set; }
+        public IDecider Decider { get; private set; }
 
```

Comment:
We can do that too, but then we need surrogate support for each and every primitive.
Because a user might decide to reference one of these primitives. e.g.

```csharp
public class MyMessage
{

        public Props MyProps {get;private set;} 
        ....
}

So the default serializer needs to be able to resolve those DTO's and turn them into the real thing even if they are embedded deep in a user message.

We already do this for `ActorRef` and `ActorPath` and maybe as you say, that is the best way to do it for all the primitives in the end, to keep the core types clean from serialization logic and extract all that into the surrogates.

So far so good, but then again, some primitives are used by other serializers that does not support surrogates.
e.g. deploy info passed around in the remoting layer carried by Protobuf.

So it's not a super easy problem to solve entirely.